### PR TITLE
Feat: Enhance get conflicts

### PR DESCRIPTION
**Pull request by carlosmondra**
This PR:

- Allows passing an info_function
- Calculates subdomains outside the info_function
- Adds a cache for domain to subdomains to speed up part of the process

### CODE REVIEW

File: tests/test_tools.py [hunk 1]
```diff
@@ -353,11 +353,11 @@ def test_create_authorid_profiles(self, client):
     def test_subdomains(self):
         # ensure that two part top-level domains are handled appropriately
         # e.g. "edu.cn", "ac.uk"
-        assert openreview.tools.subdomains('michael@mails.tsinghua.edu.cn') == ['mails.tsinghua.edu.cn', 'tsinghua.edu']
```
Comment (melisabok) [inline tests/test_tools.py:356]: why did you change the assertions here?

Comment (carlosmondra) [reply tests/test_tools.py:356]: Because the function `openreview.tools.subdomains` was doing two things" splitting the emails and then getting the subdomains for the domains of the emails. I extracted that logic and split the emails before calling the `subdomains` function

**Review by melisabok — APPROVED** [review]

**Review by celestemartinez — APPROVED** [review]
